src/net/p2p.rs

```diff
@@ -3613,2219 +3750,4119 @@ impl PeerNetwork {
+                        debug!(
+                            "{:?}: Microblock sync event {} is not connected yet",
+                            &network.local_peer, event_id
+                        );
+                        Ok((false, None))
+                    } else {
+                        // conversation died
+                        debug!(
+                            "{:?}: Microblock tip sync peer hung up",
+                            &network.local_peer
+                        );
+                        faulty_peers.dead_connections.push(event_id);
+                        Ok((true, None))
                     }
                 }
-                PeerNetworkWorkState::BlockDownload => {
-                    // go fetch blocks
-                    match dns_client_opt {
-                        Some(ref mut dns_client) => {
-                            let done = self.do_network_block_download(
-                                sortdb,
-                                mempool,
-                                chainstate,
-                                *dns_client,
-                                ibd,
-                                network_result,
-                            );
-                            if done {
-                                // advance work state
-                                self.work_state = PeerNetworkWorkState::AntiEntropy;
-                            }
-                        }
+                Some(ref mut convo) => {
+                    match convo.try_get_response() {
                         None => {
-                            // skip this step -- no DNS client available
-                            test_debug!(
-                                "{:?}: no DNS client provided; skipping block download",
-                                &self.local_peer
+                            // still waiting
+                            debug!(
+                                "{:?}: Mempool sync event {} still waiting for a response",
+                                &network.local_peer, event_id,
                             );
-                            self.work_state = PeerNetworkWorkState::AntiEntropy;
+                            // not done requesting this peer yet
+                            Ok((false, None))
+                        }
+                        Some(http_response) => {
+                            match http_response {
+                                HttpResponseType::Microblocks(_, data) => {
+                                    debug!("{:?}: Microblock sync received response with {} microblocks", &network.local_peer, data.len());
+                                    Ok((true, Some(data)))
+                                }
+                                _ => {
+                                    warn!(
+                                    "{:?}: Microblock tip sync request v2/microblocks/unconfirmed, but received unexpected response type {:?}",
```

Comment:
minor style nit: whitespace

Reply:
I'm indenting this line, I assume that's what you mean?